specification/search/resource-manager/Microsoft.Search/preview/2024-03-01-preview/search.json

```diff
@@ -0,0 +1,3298 @@
+          "$ref": "#/definitions/DataPlaneAadOrApiKeyAuthOption",
+          "description": "Indicates that either the API key or an access token from a Microsoft Entra ID tenant can be used for authentication."
+        }
+      },
+      "description": "Defines the options for how the search service authenticates a data plane request. This cannot be set if 'disableLocalAuth' is set to true."
+    },
+    "DataPlaneAadOrApiKeyAuthOption": {
+      "type": "object",
+      "properties": {
+        "aadAuthFailureMode": {
+          "type": "string",
+          "description": "Describes what response the data plane API of a search service would send for requests that failed authentication.",
+          "enum": [
+            "http403",
+            "http401WithBearerChallenge"
+          ],
+          "x-ms-enum": {
+            "name": "AadAuthFailureMode",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "http403",
+                "name": "Http403",
+                "description": "Indicates that requests that failed authentication should be presented with an HTTP status code of 403 (Forbidden)."
+              },
+              {
+                "value": "http401WithBearerChallenge",
+                "name": "Http401WithBearerChallenge",
+                "description": "Indicates that requests that failed authentication should be presented with an HTTP status code of 401 (Unauthorized) and present a Bearer Challenge."
+              }
+            ]
+          }
+        }
+      },
+      "description": "Indicates that either the API key or an access token from a Microsoft Entra ID tenant can be used for authentication."
+    },
+    "DisabledDataExfiltrationOption": {
+      "type": "string",
+      "enum": [
+        "All"
+      ],
+      "x-ms-enum": {
+        "name": "SearchDisabledDataExfiltrationOption",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "All",
+            "name": "All",
+            "description": "Indicates that all data exfiltration scenarios are disabled."
+          }
+        ]
+      },
+      "description": "A specific data exfiltration scenario that is disabled for the service."
+    },
+    "EncryptionWithCmk": {
+      "type": "object",
+      "properties": {
+        "enforcement": {
+          "description": "Describes how a search service should enforce having one or more non customer encrypted resources.",
+          "type": "string",
+          "enum": [
+            "Disabled",
+            "Enabled",
+            "Unspecified"
+          ],
+          "x-ms-enum": {
+            "name": "SearchEncryptionWithCmk",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "Disabled",
+                "name": "Disabled",
+                "description": "No enforcement will be made and the search service can have non customer encrypted resources."
```

Comment:
"can have non customer encrypted resources" is throwing me off, but I think it's just the "can have" part that makes it sound optional.  What about "No enforcement of customer-managed key encryption will be made. Only the built-in service-managed encryption is used."